redaxo/src/core/pages/system.settings.php

```diff
@@ -155,43 +169,76 @@
     $rexVersion = '<i class="rex-icon rex-icon-unstable-version" title="'. rex_i18n::msg('unstable_version') .'"></i> '. rex_escape($rexVersion);
 }
 
-$mainContent = [];
-$sideContent = [];
-$debugConfirm = '';
-
+$buttonDeleteCache = new Button(
+    slotDefault: new Slot(rex_i18n::msg('delete_cache')),
+    href: (rex_url::currentBackendPage(['func' => 'generate'] + $csrfToken->getUrlParams())),
+    variant: ButtonVariant::Danger,
+);
+
+$icon = new Icon(
+    name: IconLibrary::Debug,
+);
+$attributes = [
+    'data-pjax' => 'false',
+];
 if (!rex::isDebugMode()) {
-    $debugConfirm = ' data-confirm="' . rex_i18n::msg('debug_confirm') . '" ';
+    $attributes['data-confirm'] = rex_i18n::msg('debug_confirm');
 }
+$buttonDebugMode = new Button(
+    slotDefault: new Slot(rex::isDebugMode() ? rex_i18n::msg('debug_mode_off') : rex_i18n::msg('debug_mode_on')),
+    slotPrefix: new Slot($icon->render()),
+    href: (rex_url::currentBackendPage(['func' => 'debugmode'] + $csrfToken->getUrlParams())),
+    variant: ButtonVariant::Warning,
+    attributes: $attributes,
+);
+
+$safemodeUrl = rex_url::currentBackendPage(['safemode' => '1'] + $csrfToken->getUrlParams());
+if (rex::isSafeMode()) {
+    $safemodeUrl = rex_url::currentBackendPage(['safemode' => '0'] + $csrfToken->getUrlParams());
+}
+$buttonSaveMode = new Button(
+    slotDefault: new Slot(rex::isSafeMode() ? rex_i18n::msg('safemode_deactivate') : rex_i18n::msg('safemode_activate')),
+    href: $safemodeUrl,
+    variant: ButtonVariant::Warning,
+    attributes: [
+        'data-pjax' => 'false',
+    ],
+);
+$buttonSetup = new Button(
+    slotDefault: new Slot(rex_i18n::msg('setup')),
+    href: rex_url::currentBackendPage(['func' => 'setup'] + $csrfToken->getUrlParams()),
+    variant: ButtonVariant::Primary,
+    attributes: [
+        'data-pjax' => 'false',
+        'data-confirm' => rex_i18n::msg('setup_restart'),
+    ],
+);
 
 $content = '
     <h3>' . rex_i18n::msg('delete_cache') . '</h3>
     <p>' . rex_i18n::msg('delete_cache_description') . '</p>
-    <p><a class="btn btn-delete" href="' . rex_url::currentBackendPage(['func' => 'generate'] + $csrfToken->getUrlParams()) . '">' . rex_i18n::msg('delete_cache') . '</a></p>
+    <p>'.$buttonDeleteCache->render().'</p>
```

Comment:
Wie schon telefonisch besprochen, bin ich mir unsicher wie man beim aktuellen Konzept ein größeres Fragment notieren würde. Also ein größeres "Template", welches diverse HTML-Elemente nutzt, und innerhalb die Komponenten verwenden will.

Wir wollen ja eigentlich weg davon, dass HTML so in PHP-Strings eingebettet ist.
Und ich finde auch nicht so schön, wenn es oben einen PHP-Block gibt, wo alle Komponenten erzeugt werden, und die dann unten im HTML erst verwendet werden. Ich hätte sie lieber an Ort und Stelle.

Man könnte es zum Beispiel so in der Art notieren:

```php
    <p>
        <?= (new Button(
            slotDefault: new Slot(rex_i18n::msg('delete_cache')),
            href: (rex_url::currentBackendPage(['func' => 'generate'] + $csrfToken->getUrlParams())),
            variant: ButtonVariant::Danger,
        ))->render() ?>
    </p>
```

Ich hatte Thomas zunächst gesagt, dass ich keine `__toString`-Methode nutzen würde. Aber inzwischen überlege ich, ob man das vielleicht doch will, da die Klammern immer bisschen nerven, die man um `(new ...)` notieren muss. Wäre dann:

```php
    <p>
        <?= new Button(
            slotDefault: new Slot(rex_i18n::msg('delete_cache')),
            href: (rex_url::currentBackendPage(['func' => 'generate'] + $csrfToken->getUrlParams())),
            variant: ButtonVariant::Danger,
        ) ?>
    </p>
```